tests/unit/utils.spec.js

```diff
@@ -0,0 +1,33 @@
+import {normalizeSides} from '../../lib/utils';
+
+describe('normalizeSides', () => {
+  test.each([
+    [1, { top: 1, right: 1, bottom: 1, left: 1 }],
+    [[1, 2], { top: 1, right: 2, bottom: 1, left: 2 }],
+    [
+      { vertical: 1, horizontal: 2 },
+      { top: 1, right: 2, bottom: 1, left: 2 },
+    ],
+    [[1, 2, 3, 4], { top: 1, right: 2, bottom: 3, left: 4 }],
+    [
+      { top: 1, right: 2, bottom: 3, left: 4 },
+      { top: 1, right: 2, bottom: 3, left: 4 },
+    ],
+    [{ a: 'hi' }, { top: { a: 'hi' }, right: { a: 'hi' }, bottom: { a: 'hi' }, left: { a: 'hi' } }],
+    [
+      { vertical: 'hi' },
+      { top: 'hi', right: undefined, bottom: 'hi', left: undefined },
```

Comment:
What happens when one of the margin is undefined?

Down the code path is considered as 0?

Reply:
in the util function it would stay as undefined

- in the code it would be treated as unset and defaults to `0`

Reply:
Have now added a test case to make this clear